jaq-parse/src/filter.rs

```diff
@@ -230,6 +242,7 @@ pub fn filter() -> impl Parser<Token, Spanned<Filter>, Error = Simple<Token>> +
         path.map_with_span(|(f, path), span| Filter::path(f, path, span)),
         fold(with_comma.clone()),
         if_then_else(with_comma.clone()),
+        try_catch(sans_comma_pipe.clone()),
```

Comment:
Couldn't you just use `path` instead of `sans_comma_pipe` here?
Consider the following example:

~~~
$ jq -n 'try (try 0["0"] + 1["1"]) catch .'
"Cannot index number with string \"1\""
~~~

Here you can see that jq does not capture the addition in the inner try block, but if I understand your implementation correctly, it would capture the addition.

Reply:
Indeed! I somehow thought that the only infix operators excluded were comma and pipe. Silly me, but also maybe I got distracted by the weird "pipe error" case in jq. Hopefully this unwinds all the damage I did in the parser 😅